### PR TITLE
🤩MongoDB support 🤩

### PR DESCRIPTION
**Pull request by nikmel2803**
This PR enables MongoDB driver support for CodeX Docs. By default `local` driver is enabled (powered with [nedb](https://www.npmjs.com/package/nedb)).

Changes summary:
- created MongoDB driver (see `src/backend/database/mongodb.ts`)
- created converter from local database to mongodb (see `bin/db-converter/README.md`)
- use new type `EntityId` for all identifiers to improve support for both `ObjectId` and `string` ids.

How to run and test
1) Read `DEVELOPMENT.md`
2) Run MongoDB via docker compose
3) Run db-converter to convert your local database
4) enable mongodb driver and run application

NOTE: there are still can exists some bugs due to nedb and MongoDB compatibility issues. If you find any bugs, provide description and steps to reproduce here please

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+touch app-config.local.yaml
+```
+
+### 4. Run the application
+
+```shell
+yarn dev
+```
+
+## Starting docs with MongoDB
```
Comment (neSpecc) [inline DEVELOPMENT.md:30]: Describe that by default it will use local db

File: bin/db-converter/index.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+const pagesIdsMap = pages.reduce((acc, curr) => {
+  const newId = new ObjectId();
+
+  acc.set(curr._id, newId);
+
+  return acc;
+}, new Map());
+
+// Explicitly set the root page id
+pagesIdsMap.set('0', '0');
```
Comment (neSpecc) [inline bin/db-converter/index.js:17]: is it ok that '0' is not an `ObjectId`?

File: src/backend/controllers/pages.ts [hunk 3]
```diff
@@ -66,19 +68,34 @@ class Pages {
...
    * @param {string} pageId - pageId to exclude from result pages
    * @returns {Page[]}
    */
-  public static async groupByParent(pageId = ''): Promise<Page[]> {
-    const result: Page[] = [];
-    const orderGroupedByParent: Record<string, string[]> = {};
-    const rootPageOrder = await PagesOrder.getRootPageOrder();
-    const childPageOrder = await PagesOrder.getChildPageOrder();
-    const orphanPageOrder: PageOrder[] = [];
+  public static async groupByParent(pageId = '' as EntityId): Promise<Page[]> {
```
Comment (neSpecc) [inline src/backend/controllers/pages.ts:96]: should be tested

Comment (nikmel2803) [reply src/backend/controllers/pages.ts:96]: already tested

File: bin/db-converter/index.js (hunk 2 above)
Comment (nikmel2803) [reply bin/db-converter/index.js:17]: I don't see any problems for now with this solution. It works OK

File: DEVELOPMENT.md (hunk 1 above)
Comment (nikmel2803) [reply DEVELOPMENT.md:30]: done

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# Development guide
+
+This doc describes how to bootstrap and run the project locally.
+
+## Setup
+
+### 1. Clone the repo.
```
Comment (ilyamore88) [inline DEVELOPMENT.md:7]:
```suggestion
### 1. Clone the repo
```

**Review by ilyamore88 — APPROVED** [review]

**Review by neSpecc — APPROVED** [review]

**Pull request merged**